### DISCUSSION TITLE
Babel configuration - set up display name for styled-components

### DISCUSSION THREAD

**Discussion by topolanekmartin** [vercel/next.js#21389] [Help] [answered]
Hello,

I would want to show styled components display name in className. By default, I need to set up `.babelrc`. The configuration is shown below.

```
{
    "presets": ["next/babel"],
    "plugins": [
        [
            "styled-components",
            {
                "displayName": true,
            }
        ]
    ]
}
```

But here is the question. How can I set up this configuration only for development?

Thank you in advance for your reply.

**Comment by c0b41** [comment, marked as answer]
example usage `.babelrc.js `

```js
const dev = process.env.NODE_ENV !== 'production'

module.exports = {
  presets: ['next/babel'],
  plugins: [
    [
      'styled-components',
      {
        ssr: true,
        displayName: dev ? true : false,
        useMinify: dev ? true : false,
        pure: dev ? true : false
      }
    ]
  ]
}
```

**Reply by topolanekmartin** [reply]
That `['graphql-tag']` is redundant for me. Am I right? I don't see how it does relate with `styled-components`.

**Reply by c0b41** [reply]
> That `['graphql-tag']` is redundant for me. Am I right? I don't see how it does relate with `styled-components`.

you don't need this just my example

**Reply by topolanekmartin** [reply]
Thanks. It is working :)
